### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by zahramirkazemi** [ali-master/persian-tools#365]
**Describe the bug**
try to use `numberToWords(1000)` in your code, this bug will show up in some smartphones (for example: Samsung a11, Samsung a53,...) when the language system is **not English**.

**To Reproduce**
1. Add `numberToWords(1000)` to your code
2. Set your android smartphone language to Persian
3. Open your website

**Screenshots**
sentry screenshots:
![image](https://github.com/persian-tools/persian-tools/assets/94597090/7ec577a5-ac55-4832-9c1c-13542746d95c)
![image](https://github.com/persian-tools/persian-tools/assets/94597090/539bf346-d8e3-4c6a-a946-bab86d23a026)

**Smartphone (please complete the following information):**
 - Device: Samsung A11, Samsung A53, Samsung A14
 - OS: Android-v10
 - Browser: Chrome Mobile
 - Version: 125.0.0

**Comment by ali-master** [comment]
Thanks for reporting the issue
I'll work on it

**Comment by Faraz1995** [comment]
i have the same issue too, after four digit entered in non english language cell phone, got the same error message, i also try to convert input to english numbers but did not work for me

**Comment by mediv0** [comment]
@Faraz1995  @zahramirkazemi 

Try converting the number to English first using `digitsArToEn` or `digitsFaToEn`, then call the function. This might help.


try both functions  ArToEn FaToEn

```js
const faNum = "١";
const normal = digitsArToEn(faNum);
numberToWords(normal);
```

![image](https://github.com/persian-tools/persian-tools/assets/36822136/c53a167a-6d44-4457-8969-8bb751c4ee2b)

**Comment by mediv0** [comment]
ok so i tested it, it doesn't work with those functions i mentioned 




reproduced here:  [codesandbox](https://codesandbox.io/p/sandbox/persian-tools-issue-365-tz38pm?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyb3fqdc0006356pnmswcyc2%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyb3fqdc0002356py79bxc09%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyb3fqdc0003356pw80uav7q%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyb3fqdc0005356poq2splsq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyb3fqdc0002356py79bxc09%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyb3fqdc0001356pwign94v3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522clyb3fqdc0002356py79bxc09%2522%252C%2522activeTabId%2522%253A%2522clyb3fqdc0001356pwign94v3%2522%257D%252C%2522clyb3fqdc0005356poq2splsq%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyb3fqdc0004356prr7vdaen%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clyb3fqdc0005356poq2splsq%2522%252C%2522activeTabId%2522%253A%2522clyb3fqdc0004356prr7vdaen%2522%257D%252C%2522clyb3fqdc0003356pw80uav7q%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clyb3fqdc0003356pw80uav7q%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by elhamkarimii** [comment]
Hi @zahramirkazemi ,

Thanks for bringing up the issue.

I've noticed that the `numberToWord` function is being used with Persian numbers, which is causing the problem.
This function strictly requires English integers as input.
If we need to use Persian digits, it's our responsibility to convert them first using the `digitsFaToEn` function before passing them to `numberToWord`.

Here is a sample for handling Persian numbers:

```javascript
// First, use this function to convert the Persian number to English
const convertPersianNumber = (value: string) => digitsFaToEn(digitsArToFa(value));

// Then, use the output as input for the numberToWord function
const convertedNumber = convertPersianNumber(persianNumber);
numberToWord(convertedNumber);
```

Thank you for your attention to this matter!

**Comment by mediv0** [comment]
> Hi @zahramirkazemi ,
> 
> Thanks for bringing up the issue.
> 
> I've noticed that the `numberToWord` function is being used with Persian numbers, which is causing the problem. This function strictly requires English integers as input. If we need to use Persian digits, it's our responsibility to convert them first using the `digitsFaToEn` function before passing them to `numberToWord`.
> 
> Here is a sample for handling Persian numbers:
> 
> ```js
> // First, use this function to convert the Persian number to English
> const convertPersianNumber = (value: string) => digitsFaToEn(digitsArToFa(value));
> 
> // Then, use the output as input for the numberToWord function
> const convertedNumber = convertPersianNumber(persianNumber);
> numberToWord(convertedNumber);
> ```
> 
> Thank you for your attention to this matter!

As they mentioned above and tests i did converting digits to EN won't fix this issue. this is not about the digit ( kinda ) because it works with both FA and EN when system language is set to English

I haven't checked but it might be because of this `performer` function

https://github.com/persian-tools/persian-tools/blob/65ed6eaa26c3f310caf3ad86f9fa7bc4a50a529a/src/modules/numberToWords/index.ts#L39-L59

**Comment by zahramirkazemi** [comment]
Any update on this issue? Is it still in progress?

**Comment by ars2062** [comment]
Hi @zahramirkazemi 
As you can see I’ve already created a pull request that addresses this issue. It’s currently awaiting review and merge. Once that happens, the issue should be resolved.

You can track the progress of the PR here: #366

**Comment by zahramirkazemi** [comment]
@ars2062 

Thank you for the update and for taking care of this! I’ll monitor the PR,  Thanks for working on this!

**Comment by mediv0** [comment]
SOLVED #366

**Comment by erfanium** [comment]
Guys can you ship this soon? the latest version published on the npm is broken. Thanks!
